src/commands/image/imageSource/buildImageInAzure/UploadSourceCodeStep.ts

```diff
@@ -35,7 +40,31 @@ export class UploadSourceCodeStep extends ExecuteActivityOutputStepBase<BuildIma
         let items = await AzExtFsExtra.readDirectory(source);
         items = items.filter(i => !vcsIgnoreList.includes(i.name));
 
-        await tar.c({ cwd: source, gzip: true, file: context.tarFilePath }, items.map(i => path.relative(source, i.fsPath)));
+        await this.buildCustomDockerfileIfNecessary(context);
+        if (this._customDockerfileDirPath) {
+            // Create an uncompressed tarball with the base project
+            const tempTarFilePath: string = context.tarFilePath.replace(/\.tar\.gz/, '.tar');
+            await tar.c({ cwd: source, file: tempTarFilePath }, items.map(i => path.relative(source, i.fsPath)));
+
+            // Append/Overwrite the original Dockerfile with the custom one that was made
+            await tar.r({ cwd: this._customDockerfileDirPath, file: tempTarFilePath }, [path.relative(source, context.dockerfilePath)]);
+
+            // Create the final compressed version
+            // Note: Noticed some hanging issues when using the async version to add existing tar archives;
+            // however, the issues seem to disappear when utilizing the sync version
+            tar.c({ cwd: tmpdir(), gzip: true, sync: true, file: context.tarFilePath }, [`@${path.basename(tempTarFilePath)}`]);
+
+            try {
```

Comment:
Is this try/catch supposed to only be in the if clause? I thought we need to delete the tarFile that we create regardless of the custom logic.

Reply:
I believe the tarFile deleting comes in the `runStep` immediately after.

Reply:
In general I have some thoughts about/want some feedback regarding the try/catch structure though.  I'll reach out to discuss more offline.